talk/morelanguage/lambda.tex

```diff
@@ -86,9 +86,9 @@
   \begin{block}{First attempt in \cpp}
```

Comment:
And on a different note:
I would like to see the cppinsights as early as possible to (hopefully) demystify lambdas. What do you think?

Reply:
I am also not particularly fond of the python code, but it was there when I started working on the course and last time I taught it, one of the students wrote in the feedback that they enjoyed the python analogy made. So I think I would keep it.

Reply:
> I would like to see the cppinsights as early as possible to (hopefully) demystify lambdas. What do you think?

Hmm, I am actually fine with where we demystify lambdas. I think students should first see how lambdas look like and how they are used and then see later how the are implemented.

Reply:
I'm fine with the first Python analogy. That's indeed helpful for people who know it already. But the second one about capture only shows that there's nothing specific to be done in Python. Here I don't see how it helps, because the message is:
In C++ it's different than in Python.